extension/encoding/otlpencodingextension/go.mod

```diff
@@ -43,3 +43,23 @@ require (
 )
 
 replace github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding => ../
+
+replace github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/jaeger => ../../../pkg/translator/jaeger
+
+replace github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/zipkinencodingextension => ../zipkinencodingextension
+
+replace github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/jsonlogencodingextension => ../jsonlogencodingextension
+
+replace github.com/open-telemetry/opentelemetry-collector-contrib/pkg/golden => ../../../pkg/golden
+
+replace github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/textencodingextension => ../textencodingextension
+
+replace github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/jaegerencodingextension => ../jaegerencodingextension
+
+replace github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/zipkin => ../../../pkg/translator/zipkin
+
+replace github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal => ../../../internal/coreinternal
+
+replace github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatautil => ../../../pkg/pdatautil
+
+replace github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatatest => ../../../pkg/pdatatest
```

Comment:
I don't think any of these replaces are needed

Reply:
umm, they are due to `make crosslink`. Pipeline starts failing if I commit without the replaces.

Reply:
that might be because you have a separate package name for test, maybe

Reply:
well, if I put it in the same hierarchy as `encoding`, it throws an import cycle error.
As all other extensions import parent package. 

I think we should keep this for now and move to metadata.yaml and generate these encoding extension tests in every package, similar to what we do for other components, as suggested by @dmitryax.

Reply:
Let's write the tests manually for now in each module

Reply:
See https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/28683#discussion_r1409747527